src/stories/development/engineering_rules.stories.mdx

```diff
@@ -2,23 +2,107 @@ import { Meta } from '@storybook/addon-docs'
 
-- bad: `label-${success ? 'success' : 'error'}`
-- good: `success ? 'label-success' : 'label-error'`
-- bad: [https://css-tricks.com/the-sass-ampersand/#article-header-id-10](https://css-tricks.com/the-sass-ampersand/#article-header-id-10)
+Our main reasons for this decision:
 
-This helps to find all occurrences of classes while developing and, more importantly, [PurgeCSS](https://purgecss.com/) can detect used classes and purge the correct classes.
+1. We won’t depend on framework maintainers and their decisions.
+2. We will be flexible enough to meet any specific need of a large platform like ours.
+3. **Because we can. And we ❤️ CSS.** We are confident enough to write our stylesheets on our own.
+  CSS is _not_ some obscure, secret science that needs to be handled somewhere under the hood
+  of a 3rd-party framework. It’s a great technology that we’ve grown up with over the last
+  two decades. It’s a language we love.
+
+### CSS methodologies
+
+Not using a 3rd-party framework does not mean completely re-inventing the wheel. We are eager to
+learn what the community knows and to build upon great ideas that others brought up.
+
+#### ITCSS
+…
+
+#### BEM
+…
+
+## Tooling
+
+### CSS preprocessor
+- We use [PostCSS](https://postcss.org/).
+- We do no longer need any other (traditional) preprocessor like Sass or LESS, because:
+  - PostCSS allows for the same things we used to achieve with Sass/LESS:
+    - variables,
+    - nesting,
+    - mix-ins, functions …
+  - Relying only on PostCSS spares us some dependencies.
+  - It will reduce bundle sizes (?)
+
+### CSS-in-JS
+- For component-specific styles we employ CSS-in-JS, namely [CSS Modules](https://github.com/css-modules/css-modules).
+
+## CSS coding conventions
+
+### CSS selectors
+
+#### Classnames
+
+##### Use `kebap-case` in plain CSS
+
+As a general rule we adhere to the most commonly used style of using `kebap-case` for HTML/CSS classnames.
+
+##### Use `camelCase` in CSS-in-JS
+
+As an exception from the above rule we use `camelCase` in CSS-in-JS modules. This makes for [smoother
+integration with JavaScript](https://github.com/css-modules/css-modules#naming).
+
+##### Do not split classnames
+
+Globally used classnames should always be written in one piece, as opposed to code-wise concatenation.
+
+```javascript
+// ⛔️ don’t:
+label-${success ? 'success' : 'error'}
+
+// ✅ do:
+success ? 'label-success' : 'label-error'
+```
+
+Avoid [using preprocessors’ parent scope operators (like Sass ampersand) to concatenate classnames](
+  https://css-tricks.com/the-sass-ampersand/#aa-modifying-the-ampersand):
+
+```css
+/* ⛔️ don’t: */
+.label {
+  font-weight: normal;
+  &--heavy {
+    font-weight: bold;
+  }
+}
+
+/* ✅ do: */
+.label {
+  font-weight: normal;
+}
+
+.label--heavy {
+  font-weight: bold;
+}
+```
+
+The reason for this rule is that splitting and “crafting” classnames can make it difficult to find
+all usages of a certain class in code—which we want to avoid.
```

Comment:
```suggestion
all usages of a certain class in code — which we want to avoid.
```

Reply:
In common English 🇬🇧 typography there are [no spaces around the m-dash](https://english.stackexchange.com/a/154998). Nevertheless, we as a European team 🇪🇺  might want to settle on different conventions. That reminds me of [another story I have to fill with living content](https://github.com/betterplace/betterplace-design-system/blob/031006d2d0982b62716d36084a255b36e7dbbd75/src/stories/design/typography.stories.mdx#L10-L11) 😉 .

Reply:
Wow, didn't know that. In that case, you can leave it as is of course. My european eye reads it as one word, though. ;)

Reply:
It’s hard to read in mono-spaced source code for sure. 😅